src/gt4sd/properties/molecules/core.py

```diff
@@ -0,0 +1,238 @@
+from typing import Callable
+
```

Comment:
In general I would turn all of those in classes with `__call__` method and having parameters in the `__init__` following a common class interface. In this way all functions can be called in the same way.